runelite-client/src/main/java/net/runelite/client/ui/overlay/Overlay.java

```diff
@@ -27,21 +27,26 @@
 import java.awt.Dimension;
 import java.awt.Point;
 import java.awt.Rectangle;
+import java.util.ArrayList;
+import java.util.List;
 import lombok.Getter;
 import lombok.Setter;
 import net.runelite.client.ui.overlay.components.LayoutableRenderableEntity;
+import net.runelite.client.ui.overlay.menu.MenuOption;
 
 @Getter
 @Setter
 public abstract class Overlay implements LayoutableRenderableEntity
 {
+	private String overlayName = null;
 	private Point preferredLocation;
 	private Dimension preferredSize;
 	private OverlayPosition preferredPosition;
 	private Rectangle bounds = new Rectangle();
 	private OverlayPosition position = OverlayPosition.TOP_LEFT;
 	private OverlayPriority priority = OverlayPriority.NONE;
 	private OverlayLayer layer = OverlayLayer.UNDER_WIDGETS;
+	private List<MenuOption> menuOptions = new ArrayList<>();
```

Comment:
Why you have both buildMenuOptions and this

Reply:
This stores the menuOptions that were displayed at the time of opening the right-click menu. The MenuEntries store the Overlay Index `Param0` and MenuOption index `Param1` so we can reference the right runnable later on.

Since the results of `buildMenuOptions` could be different than the options displayed in the menu i figured storing it was the best solution.

Reply:
Well, that seems wrong. If you have to result to something like this, then I would simply not use the `buildMenuEnries` method at all and simply use just the menuOptions array.

Reply:
I thought there might be use cases where certain menu options should only be added in specific cases. Just having the array will require manually grabbing and adjusting the array which seems like a lot of duplicate code inside overlays.

I'll make the changes as requested it just seems to be a step backwards IMO.